refactor: move templates shared stuff to lib folder

Letushev opened the pull request.

yurii-ve commented on _codux/boards/color-select.board.tsx:
```diff
@@ -1,6 +1,6 @@
 import { useState } from 'react';
 import { createBoard } from '@wixc3/react-board';
-import { ColorSelect, ColorSelectOption } from '~/components/color-select/color-select';
+import { ColorSelect, ColorSelectOption } from 'lib/components/color-select/color-select';
```
not sure about moving components to the lib.. might be confusing for the user why some components are in `lib` and some in `src`. For example `Accordion` sounds like generic component and it's in the `src`, on the other hand - `RangeSlider`, also looks like a generic component but it's in the `lib`.

Letushev commented:
@alisey @yurii-ve @olehrakwix PR is ready for review
- Renamed `lib/api` to `lib/ecom`
- Moved `useCart` to `lib/ecom/hooks`
- Changed functions to get data in route loaders
- Added index files to `lib` subfolders except for `lib/components`

alisey commented on _codux/board-wrappers/component-wrapper.tsx:
```diff
@@ -1,6 +1,6 @@
 import { createRemixStub } from '@remix-run/testing';
 import { PropsWithChildren } from 'react';
-import { EcomAPIContextProvider } from '~/api/ecom-api-context-provider';
+import { EcomAPIContextProvider } from 'lib/ecom';
```
Oh, I didn't realize imports are going to look like this, `lib` looks like a name of an NPM package. For some reason I though it's going to be `~/lib/ecom`.

Maybe we should do this?
`~/src/components`
`~/lib/components`

alisey replied:
Even without `lib`, `~/src/router` is more intuitive for me than `~/router`

Letushev replied:
Updated all imports to the pattern
`~/src/components`
`~/lib/components`
